docs/proposals/03-loadbalancer.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Network Loadbalancer
+
+oep-number: 3
+
+creation-date: 2022-10-18
+
+status: implementable
+
+authors:
+
+- "@gehoern"
+reviewers:
+- "@adracus"
+- "@afritzler"
+- "@guvenc"
+- "@MalteJ"
+
+---
+
+# OEP-3: Network Loadbalancer
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+    - [Details](#details)
+- [Proposal](#proposal)
+
+## Summary
+Network service for Load Balancing is needed in every modern infrastructure (e.g. Kubernetes). Load balancing makes services scalable, fault-tolerant and accessible from external users. The load balancer described here is a L3-Loadbalancer (so it implements an IP level load balancing). The load balancer function is a standard network function that can be to be deployed (as an implementation detail) on a platform/node that is capable of running network functions (e.g. dp-service enabled nodes/nic's) 
+
+## Motivation
+The onMetal network is a fully routed network. To run workloads hosted on several nodes (in Kubernetes terms equivalent to cluster IP's) and not only on a single node (node IP's) solutions like ECMP (Equal cost multi pathing) seem to be the natural solution. But ECMP has certain limitations: the amount of nodes it can be used to distribute traffic to and a hash based algorithm to pin the network flows, that changes - based on the amount of targets - its distributed pinning. 
```

Comment:
The motivation here goes into the direction of how an LB can be implemented. In `onmetal-api` we should rather focus on the general motivation why Load Balancers are necessary. In our case e.g., we have the `VirtualIP` which is a single stable public IP attached to a `NetworkInterface`, however, to scale beyond single network interfaces, `LoadBalancer`s offer the capability to dynamically route traffic to multiple targets.

Reply:
I believe, yeah, considerations regarding exact implementation should be moved to the project that will handle the provisioning of load balancer service, e.g. `lblet`.

Also, it is not clear from the motivation whether ECMP will be used or not, which features would be included (e.g. will L3DSR be supported), etc.

Reply:
tried to make it less specific but still describing the basic idea. Took your sentence @adracus